### PR TITLE
tests: subsys: modbus cases move to new ztest API

### PR DESCRIPTION
**Pull request by Zhaoningx**
Move test to use new ztest API.

Signed-off-by: NingX Zhao <ningx.zhao@intel.com>

### CODE REVIEW

File: tests/subsys/modbus/src/main.c [hunk 1]
```diff
@@ -6,54 +6,69 @@
...
+	test_coil_wr_rd();
+	test_di_rd();
+	test_input_reg();
+	test_holding_reg();
+	test_diagnostic();
+	test_client_disable();
+	test_server_disable();
+}
+
+ZTEST_SUITE(modbus_client, NULL, NULL, NULL, NULL, NULL);
```
Comment (jfischer-no) [inline tests/subsys/modbus/src/main.c:74]: It is not client specific, please _s/modbus_client/modbus/g_

`ZTEST_SUITE(modbus, NULL, NULL, NULL, NULL, NULL);`

**Review by jfischer-no — CHANGES_REQUESTED** [review]

File: tests/subsys/modbus/src/main.c (hunk 1 above)
Comment (Zhaoningx) [reply tests/subsys/modbus/src/main.c]: Updated, thanks.

**Review by enjiamai — APPROVED** [review]

**Comment by enjiamai** [comment]
Hi, @yperess @aaronemassey , could you please help on this PR:  is there a better to organize this tests? Thanks!

**Review by aaronemassey — APPROVED** [review]
So as far I'm concerned this is fine. We have a single suite, for a particular feature and we're using helpers within the test to test behavior and state specific to each test.

So in terms of the ZTEST API, this looks fine to me. It's also more clear in the new API what each test is really doing versus when all the tests were just a single list.

Thank you!

**Review by jfischer-no — APPROVED** [review]

**Pull request merged**